### ISSUE TITLE
Rule request: Prefer key path expressions over closures

### ISSUE THREAD

**Issue by marcelofabri** [realm/SwiftLint#3157]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### New rule request

Swift 5.2 adds [SE-0249](https://github.com/apple/swift-evolution/blob/master/proposals/0249-key-path-literal-function-expressions.md), which allows key path expressions to be used as functions. 

1. Why should this rule be added? Share links to existing discussion about what
   the community thinks about this.

https://www.hackingwithswift.com/articles/212/whats-new-in-swift-5-2

2. Provide several examples of what _would_ and _wouldn't_ trigger violations.

```swift
// OK
users.map(\.email)
users.filter(\.isAdmin)
users.map { $0.email() }
users.map { user in 
  user.email()
}

// Triggers
users.map { $0.email }
users.map { user in 
  user.email
}
```

3. Should the rule be configurable, if so what parameters should be configurable?

Just severity

4. Should the rule be opt-in or enabled by default? Why?
   See [README.md](../README.md#opt-in-rules) for guidelines on when to mark a rule as opt-in.

Opt-in, because it's only Swift 5.2+.

**Comment by marcelofabri** [comment]
I've started working on this, but turns out that Xcode 11.4 doesn't autocomplete keypath expressions in that context, so I'm dropping it for now. If anyone wants to pick up, I have a very WIP branch: https://github.com/realm/SwiftLint/compare/mf-keypath-over-closure?expand=1

**Comment by marcelofabri** [comment]
Spent some more time on this today since Xcode 12 will autocomplete keypath expressions, but it doesn't look very promising. This would be much easier with SwiftSyntax and `MemberAccessExpr`.

**Comment by marcelofabri** [comment]
Duplicate of #3082
